adapters/beachfront/beachfront.go

```diff
@@ -197,182 +188,278 @@ func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
-		beachfrontReqs.Banner, errs = getBannerRequest(request)
+	if wantsBanner {
+		beachfrontReqs.Banner, errs = getBannerRequest(request, errs)
 	}
 
 	if len(videoImps) > 0 {
-		var videoErrs []error
-		var videoList []beachfrontVideoRequest
+		requestStub := *request
+		requestStub.Imp = nil
 
-		request.Imp = videoImps
+		beachfrontReqs.NurlVideo, beachfrontReqs.ADMVideo, errs = getVideoRequests(requestStub, videoImps, errs)
+	}
 
-		videoList, videoErrs = getVideoRequests(request)
-		errs = append(errs, videoErrs...)
+	return beachfrontReqs, errs
+}
 
-		for i := 0; i < len(videoList); i++ {
-			if videoList[i].VideoResponseType == "nurl" || videoList[i].VideoResponseType == "both" {
-				beachfrontReqs.NurlVideo = append(beachfrontReqs.NurlVideo, videoList[i])
-			}
+func getBannerRequest(request *openrtb.BidRequest, errs []error) (bannerRequest, []error) {
+	var secure int8
+	var bannerReq bannerRequest
+	bannerReq, secure, errs = impsToSlots(request.Imp, errs)
 
-			if videoList[i].VideoResponseType == "adm" || videoList[i].VideoResponseType == "both" {
-				beachfrontReqs.ADMVideo = append(beachfrontReqs.ADMVideo, videoList[i])
-			}
+	if len(bannerReq.Slots) == 0 {
+		errs = append(errs, errors.New("unable to construct a valid banner request"))
+		return bannerReq, errs
+	}
+
+	if request.Site != nil && request.Site.Page != "" {
+		bannerReq.Secure = isPageSecure(request.Site.Page)
+	} else {
+		bannerReq.Secure = secure
+	}
+
+	if request.Device != nil {
+		bannerReq.IP = getIP(request.Device.IP)
+		bannerReq.DeviceModel = request.Device.Model
+		bannerReq.DeviceOs = request.Device.OS
+		if request.Device.DNT != nil {
+			bannerReq.Dnt = *request.Device.DNT
+		}
+		if request.Device.UA != "" {
+			bannerReq.UA = request.Device.UA
 		}
 	}
 
-	return
-}
+	var t = fallBackDeviceType(request)
 
-func getAppId(ext openrtb_ext.ExtImpBeachfront, media openrtb_ext.BidType) (string, error) {
-	var appid string
-	var error error
+	if t == openrtb.DeviceTypeMobileTablet {
+		bannerReq.Page = request.App.Bundle
+		if request.App.Domain == "" {
+			bannerReq.Domain = getDomain(request.App.Domain)
+		} else {
+			bannerReq.Domain = request.App.Domain
+		}
 
-	if fmt.Sprintf("%s", reflect.TypeOf(ext.AppId)) == "string" &&
-		ext.AppId != "" {
+		bannerReq.IsMobile = 1
+	} else if t == openrtb.DeviceTypePersonalComputer {
+		bannerReq.Page = request.Site.Page
+		if request.Site.Domain == "" {
+			bannerReq.Domain = getDomain(request.Site.Page)
+		} else {
+			bannerReq.Domain = request.Site.Domain
+		}
 
-		appid = ext.AppId
-	} else if fmt.Sprintf("%s", reflect.TypeOf(ext.AppIds)) == "openrtb_ext.ExtImpBeachfrontAppIds" {
-		if media == openrtb_ext.BidTypeVideo && ext.AppIds.Video != "" {
-			appid = ext.AppIds.Video
-		} else if media == openrtb_ext.BidTypeBanner && ext.AppIds.Banner != "" {
-			appid = ext.AppIds.Banner
+		bannerReq.IsMobile = 0
+	}
+
+	if request.User != nil && request.User.ID != "" {
+		if bannerReq.User.ID == "" {
```

Comment:
Is it possible for `bannerReq.User.ID` to have a value at this point?

Reply:
No it is no. I've fixed that locally.